File: C++/题目1202：排序.cpp

// 题目1202：排序

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:54
 * @url:http://ac.jobdu.com/problem.php?pid=1202
 */

import java.util.Arrays;
#include<stdio.h>

const int maxn = 102;
int array[maxn];
int n, i;

int sortUnit(int low, int high) {
    int key = array[low];
    while (low < high) {
        while (low < high && array[high] >= key) {
            high--;
        }
        array[low] = array[high];
        while (low < high && array[low] <= key) {
            low++;
        }
        array[high] = array[low];
    }
    array[high] = key;
    return high;
}

void sort(int low, int high) {
    if (low > high) {
        return;
    }
    int index = sortUnit(low, high);
    sort(low, index - 1);
    sort(index + 1, high);
}

int main() {
    while (scanf("%d", &n) != EOF) {
        for (i = 0; i < n; i++) {
            scanf("%d", &array[i]);
        }
        sort(0, n - 1);
        for (i = 0; i < n; i++) {
            printf("%d ", array[i]);
        }
        printf("\n");
    }
    return 0;
}
/**************************************************************
Problem: 1202
User: wangzhenqing
Language: C++
Result: Accepted
Time:30 ms
Memory:1020 kb
****************************************************************/


                        

File: Java/题目1202：排序.java

// 题目1202：排序

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:54
 * @url:http://ac.jobdu.com/problem.php?pid=1202
 */

import java.util.Arrays;
import java.util.Scanner;

/** 
 * @ClassName: Main1046 
 * @Description: TODO 
 * @author wangzq
 * @date 2013-8-7 下午04:00:39 
 * 
 * @version 3.0.0 
 */
public class Main {

    /*
     * 1202
     */
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (scanner.hasNext()) {
            int n = scanner.nextInt();
            int array[] = new int[n];
            for (int i = 0; i < n; i++) {
                array[i] = scanner.nextInt();
            }
            Arrays.sort(array);
            String result = "";
            for (int i = 0; i < array.length; i++) {
                result += array[i] + " ";
            }
            System.out.println(result);
        }
    }
}

/**************************************************************
 Problem: 1202
 User: wzqwsrf
 Language: Java
 Result: Accepted
 Time:720 ms
 Memory:95060 kb
 ****************************************************************/




